technical-docs/designs/infra/relativepaths.md

```diff
@@ -0,0 +1,240 @@
+
+Another option is to copy modules from cached remote directory to the generated
+folder. This allows SPK to directly reference the parent module source with the
+generated terraform templates. In addition, this would also require the
+templates to be placed accordingly in respects to relative paths to parent
+modules. In Bedrock, the template folders are 3 levels down
+(../../../cluster/azure/keyvault_flex_vol")
+
+```
+fabrikam/
+    definition.yaml
+    fabrikam-central/
+        definition.yaml
+    fabrikam-east/
+        definition.yaml
+    fabrikam-west/
+        definition.yaml
+fabrikam-generated/
+    cluster
+        azure
+        common
+        minikube
+        environments
+            deployments
+                fabrikam-central/
+                    main.tf
+                    variables.tf
+                    keyvault.tf
+                    terraform.tfvars
+                    backend.tfvars
+                fabrikam-east/
+                    main.tf
+                    variables.tf
+                    keyvault.tf
+                    terraform.tfvars
+                    backend.tfvars
+                fabrikam-west/
+                    main.tf
+                    variables.tf
+                    keyvault.tf
+                    terraform.tfvars
+                    backend.tfvars
+```
+
+Limitations:
+
+- Very coupled to current organization of Bedrock and will have breaking changes
+  when introducing new folders at template /module levels
+- Copying modules twice local to agent, once for cache and again during generate
+
+### 2.3 Use a symlink to reference the modules
+
+A symbolic link, also termed a soft link, is a special kind of file that points
+to another file, much like a shortcut in Windows or a Macintosh alias. This will
+allow generate the alias the cached repo for all `spk infra generate` commands.
+
+`ln -s modules /path/to/modules`
+
+Inside of spk, the terraform files will now reference the symlink which is
+cached.
+
+**Output `Main.tf`**
+
+```
+terraform {
+  backend "azurerm" {}
+}
+
+data "azurerm_client_config" "current" {}
+
+data "azurerm_resource_group" "cluster_rg" {
+  name = var.resource_group_name
+}
+
+data "azurerm_resource_group" "keyvault" {
+  name = var.keyvault_resource_group
+}
+
+data "azurerm_subnet" "vnet" {
+  name                 = var.subnet_name
+  virtual_network_name = var.vnet_name
+  resource_group_name  = data.azurerm_resource_group.keyvault.name
+}
+
+module "aks-gitops" {
+  source = "modules/azure/aks-gitops"
+}
+
+# Create Azure Key Vault role for SP
+module "keyvault_flexvolume_role" {
+  source = "modules/azure/keyvault_flexvol_role"
+}
+
+# Deploy central keyvault flexvolume
+module "flex_volume" {
+  source = "modules/azure/keyvault_flexvol"
+}
+```
+
+Limitations:
+
+- Modifying the terraform output files.
+- Added complexity with symlink alias which prevents ease of simply running
+  terraform commands on native machine
```

Comment:
@NathanielRose looks good. Can we add sections for 
```
3. Dependencies
4. Risks & Mitigations
5. Documentation
6. Alternatives (move the 2.2. and 2.3 here)
```

Reply:
@andrebriggs Completed.